src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Services/NuGetSourcesService.cs

```diff
@@ -93,6 +93,7 @@ private IReadOnlyList<PackageSource> GetPackageSourcesToUpdate(IReadOnlyList<Pac
                     if (packageSource.Name.Equals(packageSourceContextInfo.Name, StringComparison.InvariantCulture)
                         && packageSource.Source.Equals(packageSourceContextInfo.Source, StringComparison.InvariantCulture)
                         && packageSource.ProtocolVersion == packageSourceContextInfo.ProtocolVersion
+                        && packageSource.AllowInsecureConnections == packageSourceContextInfo.AllowInsecureConnections
```

Comment:
The comment above, which I don't think is useful, is now inaccurate.
1. `// If key properties have not changed, we don't need to do anything`
2. `// If identifying properties have not changed, we don't need to do anything`
3. `// If Name/Source/IsEnabled/ProtocolVersion/AllowInsecureConnections has not changed, we don't need to do `anything
4. delete the comment

I vote for 4 or 1.